fvm/src/executor/default.rs

```diff
@@ -109,16 +110,29 @@ where
                     gas_used,
                 }
             }
-            Err(ExecutionError::Syscall(SyscallError(errmsg, exit_code))) => {
-                if exit_code.is_success() {
-                    return Err(anyhow!(
-                        "message invocation errored with an ok status: {}",
-                        errmsg
-                    ));
-                }
+            Err(ExecutionError::OutOfGas) => Receipt {
+                exit_code: ExitCode::SysErrOutOfGas,
+                return_data: Default::default(),
+                gas_used,
+            },
+            Err(ExecutionError::Syscall(SyscallError(errmsg, err_code))) => {
+                let exit_code = match err_code {
+                    ErrorNumber::IllegalOperation => ExitCode::SysErrIllegalActor,
+                    ErrorNumber::AssertionFailed => ExitCode::SysErrIllegalArgument,
+                    ErrorNumber::InsufficientFunds => ExitCode::SysErrInsufficientFunds,
+                    ErrorNumber::NotFound => ExitCode::SysErrInvalidReceiver,
+                    code => {
+                        return Err(anyhow!(
+                            "unexpected syscall error when processing message: {} ({})",
+                            code,
+                            code as u32
+                        ))
+                    }
+                };
+
                 backtrace.push(CallError {
                     source: 0,
-                    code: exit_code,
+                    code: CallErrorCode::Syscall(err_code),
```

Comment:
Ah, yeah. This was from before the syscall error code -> error number rename.

Reply:
So, outcome isn't correct. This is the _code_ attached to an error (either syscall or exit).

But I don't actually need this enum anymore. Instead, I can have:

1. An optional "cause" (the syscall error that caused the first actor to exit).
2. A backtrace of actor exits.

But I'm going to merge the current PR first then iterate because I want to make some other changes as well.